### ISSUE TITLE
Unpredictable behaviuor of function ilike.

### ISSUE THREAD

**Issue by Melchizedek13** [ClickHouse/ClickHouse#16890]
Hi everyone!

Today I came across an amazing feature in the `ilike(...)` function, which I would like to demonstrate to you with the following query:
```sql
  with '%insert into data_marts.client_rfm%''app''%' as pattern
select lower(query) like pattern like_res,
       query ilike pattern       ilike_res
  from values('query String',
       'INSERT INTO data_marts.client_rfm (...) 
          WITH CAST(''2020-11-08'' AS Date) AS report_date
        SELECT report_date
             , ''app''
             , city'
  )
;
```
Column results differ from each other.
As far as I understand, this is unacceptable behaviour in the function `ilike(...)`. The results should be the same.
![image](https://user-images.githubusercontent.com/10761242/98832544-00899e00-244e-11eb-95d5-7f872a349c26.png)

Moreover, I also noticed that the presence of sorting and limitation affects the result of the function `ilike(...)`
I would like to demonstrate the difference in how the function works with some examples below.

without limit & order by:
![image](https://user-images.githubusercontent.com/10761242/98832095-73dee000-244d-11eb-8cf2-e7cbdb5ed503.png)

with limit & order by:
![image](https://user-images.githubusercontent.com/10761242/98832246-a557ab80-244d-11eb-9b1b-8590388bf58b.png)

with order by only:
![image](https://user-images.githubusercontent.com/10761242/98832444-e354cf80-244d-11eb-8404-bd944398af2e.png)

with limit only:
![image](https://user-images.githubusercontent.com/10761242/98832488-f23b8200-244d-11eb-8b20-3944e53ae6a4.png)

I use ClickHouse 20.8.3.18 version.

Could you explain the behaviour of function `ilike(...)`?

Thanks in advance.

**Comment by filimonov** [comment]
It seems fixed in 20.8.4 and newer...

**Comment by filimonov** [comment]
I guess it fixed by https://github.com/ClickHouse/ClickHouse/pull/15536 
Please update your ClickHouse version.
